src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+  }),
+  optionPropType,
+]);
+
+const OptionDefault = React.createClass({
+
+  displayName: 'Option',
+
+  propTypes: {
+    option: PropTypes.shape({
+      ...optionPropType,
+      label: PropTypes.string.isRequired,
+    }).isRequired,
+  },
+
+  render() {
+    let {
+      option: {label},
+      ...otherProps,
+    } = this.props;
+
+    return (
+      <div className="Option" {...otherProps}>
+        <span>{label}</span>
+      </div>
+    );
+  },
+
+});
+
+const Select = React.createClass({
+
+  displayName: 'Select',
+
+  mixins: [
+    InputMixin(optionPropType),
+  ],
+
+  propTypes: {
+    className: PropTypes.string,
+    placeHolder: PropTypes.string,
+    options: PropTypes.arrayOf(optionGroupOf(optionPropType)).isRequired,
+    optionRender: PropTypes.func,
+    filterOption: PropTypes.func, //By default filter option by their label.
+    hideGroupsWithNoMatch: PropTypes.bool,
+    valueLink: PropTypes.shape({
+      value: optionPropType,
+      requestChange: PropTypes.func.isRequired,
+    }).isRequired,
+  },
+
+  //Life Cycle methods
+
+  getDefaultProps() {
+    return {
+      placeHolder: DEFAULT_PLACEHOLDER,
+      optionRender: OptionDefault,
+      hideGroupsWithNoMatch: true,
+      filterOption: (filter, option, group) => (new RegExp(filter, 'i').test(option.label)),
+    };
+  },
+
+  getInitialState() {
+    return {
+      filterValue: '',
+      suggestedOption: null,
+      isFocused: false,
+      openGroupsId: [],
+    };
+  },
+
+  //Prop Helpers: value
+
+  _removeSelection() {
+    this.requestChange({ $set: null });
+  },
+
+  _selectOption(option) {
+    this.requestChange({ $set: option });
+    this._clearFilterValue();
+    this._closeList();
+    this._blurFilterInput();
+  },
+
+  //State Helpers: openGroupsId
+
+  _isGroupOpen(openGroupsId, group) {
+    return _.contains(openGroupsId, group.id);
+  },
+
+  _toggleGroupOpenState(group) {
+    let { openGroupsId } = this.state,
+        isOpened = this._isGroupOpen(openGroupsId, group);
+
+    if (isOpened) {
+      this._closeGroup(group);
+    } else {
+      this._openGroup(group);
+    }
+  },
+
+  _openGroup(group) {
+    let { openGroupsId } = this.state;
+    this.setState({ openGroupsId: update(openGroupsId, {$push: [group.id]}) });
+  },
+
+  _closeGroup(group) {
+    let { openGroupsId } = this.state;
+    let index = _.findIndex(openGroupsId, (id) => id === group.id);
+    this.setState({ openGroupsId: update(openGroupsId, {$splice: [[index, 1]]}) });
+  },
+
+  _openSuggestedGroup(suggestionOption) {
+    if (!suggestionOption) {
+      return;
+    }
+    let { options } = this.props,
+        { openGroupsId } = this.state;
+
+    let suggestedGroup = _.find(options, (group) => group.isGroup && _.contains(_.pluck(group.options, 'id'), suggestionOption.id));
+    if (suggestedGroup && !this._isGroupOpen(openGroupsId, suggestedGroup)) {
+      this._openGroup(suggestedGroup);
+    }
+  },
```

Comment:
However, right now it looks like closing the suggested option's group closes the select window.